run.py

```diff
@@ -155,6 +180,23 @@ def add_data_link(session, current_row):
     result = session.execute(q)
     current_row['data_links_rows'] = fetch_data_link_record(result.fetchall())
 
+def is_article(property):
+    """is the passed property value for an article or nonarticle
+
+    passed an array of strings, the current property value for the bibcode"""
+    article_types = set(['eprint', 'article', 'inproceedings', 'inbook'])
```

Comment:
a small detail, but since you are going to call this many millions times, article_types should be initialized once